apache/zeppelin
# [ZEPPELIN-2427] Hide navbar and noteAction bar when scrolling down

### What is this PR for?

Hide navbar and noteAction bar when scrolling down to 

- **widen view area**

Here are few famous services which hiding their navbar too.

- https://www.kaggle.com/c/titanic
- http://github.com/tj (profile page)

### Implementation Details

Since [headroom](wicky.nillia.ms/headroom.js/) functionality makes integration tests fail,
we can't use `headroom` for CI build. to resolve this issue, 

- created new maven profiles: `web-ci` and `web-dist` (default)
- maven will execute `yarn run build:ci` with the `web-ci` profile which doesn't include headroom in `app.js`

### What type of PR is it?
[Improvement]

### Todos
* [x] - Add headroom.js
* [x] - Filter out useless dependencies
* [x] - Apply headroom.js

### What is the Jira issue?

[ZEPPELIN-2427](https://issues.apache.org/jira/browse/ZEPPELIN-2427)

### How should this be tested?

1. Open Zeppelin
2. Scroll down, up to see whether navbar and note action bar are hided or not.

### Screenshots (if appropriate)

![2427](https://cloud.githubusercontent.com/assets/4968473/25214004/afae7a1c-25cf-11e7-9ee1-f41e767d2039.gif)


### Questions:
* Does the licenses files need update?
* Is there breaking changes for older versions?
* Does this needs documentation?
